dart/lib/src/sentry_tracer.dart

```diff
@@ -0,0 +1,104 @@
+import '../sentry.dart';
+import 'tracing.dart';
+
+class SentryTracer extends ISentrySpan {
+  final Hub _hub;
+  late String _name;
+
+  // missing waitForChildren
+
+  late ISentrySpan _rootSpan;
+  final List<ISentrySpan> _children = [];
+
+  SentryTracer(SentryTransactionContext transactionContext, this._hub) {
+    _rootSpan = SentrySpan(this, transactionContext);
+    _name = transactionContext.name;
+  }
+
+  @override
+  Future<void> finish({SpanStatus? status}) async {
+    await _rootSpan.finish(status: status);
+    await captureTransaction();
+  }
+
+  @override
+  void removeData(String key) {
+    _rootSpan.removeData(key);
+  }
+
+  @override
+  void removeTag(String key) {
+    _rootSpan.removeTag(key);
+  }
+
+  @override
+  void setData(String key, value) {
+    _rootSpan.setData(key, value);
+  }
+
+  @override
+  void setTag(String key, String value) {
+    _rootSpan.setTag(key, value);
+  }
+
+  @override
+  ISentrySpan startChild(
+    String operation, {
+    String? description,
+  }) {
+    return _rootSpan.startChild(
+      operation,
+      description: description,
+    );
+  }
+
+  ISentrySpan startChildWithParentId(
+    SpanId parentId,
+    String operation, {
+    String? description,
+  }) {
+    final context = SentrySpanContext(
+      traceId: _rootSpan.context.traceId,
+      parentId: parentId,
+      operation: operation,
+      description: description,
+      sampled: _rootSpan.context.sampled,
+    );
+
+    final child = SentrySpan(this, context);
+
+    _children.add(child);
+
+    return child;
+  }
+
+  // missing hasUnfinishedChildren & isWaitingForChildren feature
+
+  Future<void> captureTransaction() async {
+    _hub.configureScope((scope) {
+      if (scope.span == this) {
+        scope.span = null;
+      }
+    });
+
+    final transaction = _toTransaction();
+    await _hub.captureTransaction(transaction);
+  }
+
+  SentryTransaction _toTransaction() {
+    // filter unfinished spans _children
```

Comment:
add filtering of spans

Reply:
lets finish them with `timeline_exceed` and use the timestamp from the transaction timestamp

Reply:
typo: deadline_exceeded